pkg/gitprovider/gitea.go

```diff
@@ -420,3 +423,126 @@ func (g *GiteaGitProvider) GetDefaultBranch(staticContext *StaticGitContext) (*s
 
 	return &repo.DefaultBranch, nil
 }
+
+func (g *GiteaGitProvider) RegisterPrebuildWebhook(repo *GitRepository, endpointUrl string) (string, error) {
+	client, err := g.getApiClient()
+	if err != nil {
+		return "", fmt.Errorf("failed to get api client: %w", err)
+	}
+
+	hookOpts := gitea.CreateHookOption{
+		Type: "gitea",
+		Config: map[string]string{
+			"url":          endpointUrl,
+			"content_type": "json",
+		},
+		Events: []string{"push"},
+		Active: true,
+	}
+
+	hook, _, err := client.CreateRepoHook(repo.Owner, repo.Name, hookOpts)
+	if err != nil {
+		return "", fmt.Errorf("failed to create webhook: %w", err)
+	}
+
+	return strconv.Itoa(int(hook.ID)), nil
+}
+
+func (g *GiteaGitProvider) GetPrebuildWebhook(repo *GitRepository, endpointUrl string) (*string, error) {
+	client, err := g.getApiClient()
+	if err != nil {
+		return nil, err
+	}
+
+	hooks, _, err := client.ListRepoHooks(repo.Owner, repo.Name, gitea.ListHooksOptions{
+		ListOptions: gitea.ListOptions{
+			PageSize: 100,
+			Page:     1,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, hook := range hooks {
+		if hook.Config["url"] == endpointUrl {
+			return util.Pointer(strconv.Itoa(int(hook.ID))), nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (g *GiteaGitProvider) UnregisterPrebuildWebhook(repo *GitRepository, id string) error {
+	client, err := g.getApiClient()
+	if err != nil {
+		return err
+	}
+
+	hookId, _ := strconv.Atoi(id)
```

Comment:
Let's not ignore the error here. You never know :)

Reply:
True, they pop up when you least expect them! 👍